pkg/connector/repository.go

```diff
@@ -263,6 +264,12 @@ func (o *repositoryResourceType) Grant(ctx context.Context, principal *v2.Resour
 		return nil, err
 	}
 
+	enIDParts := strings.Split(en.Id, ":")
```

Comment:
LGTM. but, I noticed we have the same code twice, It would be a good idea to reuse this code and add a helper or a function.  :)